tmol/score/ref/ref_energy_term.py

```diff
@@ -0,0 +1,71 @@
+import torch
+
+from ..energy_term import EnergyTerm
+
+from tmol.database import ParameterDatabase
+from tmol.score.ref.ref_whole_pose_module import RefWholePoseScoringModule
+
+from tmol.chemical.restypes import RefinedResidueType
+from tmol.pose.packed_block_types import PackedBlockTypes
+from tmol.pose.pose_stack import PoseStack
+
+
+class RefEnergyTerm(EnergyTerm):
+    device: torch.device  # = attr.ib()
+
+    def __init__(self, param_db: ParameterDatabase, device: torch.device):
+        super(RefEnergyTerm, self).__init__(param_db=param_db, device=device)
+
+        self.ref_weights = param_db.scoring.ref.weights
+        self.device = device
+
+    @classmethod
+    def score_types(cls):
+        import tmol.score.terms.ref_creator
+
+        return tmol.score.terms.ref_creator.RefTermCreator.score_types()
+
+    def n_bodies(self):
+        return 1
+
+    def setup_block_type(self, block_type: RefinedResidueType):
+        super(RefEnergyTerm, self).setup_block_type(block_type)
+
+        if hasattr(block_type, "ref_weight"):
+            return
+
+        ref_weight = self.ref_weights[
+            block_type.base_name
+        ]  # TODO: 0.0 or throw error if key not found?
```

Comment:
@fdimaio 

What should be the behavior here? Default to zero or throw an exception?